docs/kore-syntax.md

```diff
@@ -0,0 +1,143 @@
+Syntax of the KORE language
+===========================
+
+This documents describes the syntax of KORE, as admitted by the Haskell Backend.
+
+It is meant to be an easier to access source than the
+[Semantics of K](docs/semantics-of-k.pdf)
+
+Any change in the KORE Parser infrastructure should be reflected here.
+
+Lexical Syntax
+--------------
+
+### Comments
+
+- C-style comments (non-nested)
+
+### Strings
+
+```
+StringLiteral ::= "\"" {Char}* "\""
```

Comment:
It's fairly obvious, but, if it's not hard, I'd add a link to something describing the syntax used here, e.g. that things between quotation marks are literals, curly braces are used to identify things to which "repeaters" apply (e.g. `+*?`), and so on.

Reply:
I've changed it to valid BNF and added a link to wikipedia